### PR TITLE
Add bill command

### LINKED ISSUE #81: As a hotel manager, I want to view the bill of the hotel guest
...so that I can charge the guest the right amount

### LINKED ISSUE #82: As a hotel manager, I want to be able to add to the bill of the hotel guest
...so that I can update it if there are any changes

### LINKED ISSUE #83: As a hotel manager, I want to be able to deduct from the bill of the hotel guest
...so that I can update it if there are any changes

### LINKED ISSUE #99: Add Bill command
- Syntax: bill INDEX (+/-)VALUE
- New value must be non-negative

### PR DESCRIPTION
**Pull request by JustinPeng13**
## Added
- BillCommand class
- BillCommandParser class
- BillCommandTest class
- BillCommandParserTest class
- Bill command in user guide
- Bill prefix "b/"
- Add method for bill
## Modified
- Bill accepts negative values but not for final bill
- Test data
- Sample data
## Comments
- Bill command and bill class can accept +, -, or unsigned (taken as positive) values
- Bill is checked for negative value before being assigned to guest

Resolves #99, resolves #82, resolves #83, resolves #81

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -121,6 +123,26 @@ Examples:
 *  `edit 1 p/91234567 e/johndoe@example.com` Edits the phone number and email address of the 1st guest to be `91234567` and `johndoe@example.com` respectively.
 *  `edit 2 n/Betsy Crower` Edits the name of the 2nd guest to be `Betsy Crower`.
 
+### Billing a guest : `bill`
+
+Updates the bill of a guest in the guest book.
+
+Format: `bill INDEX b/BILL`
```
Comment (jehousoh) [inline docs/UserGuide.md:130]: Should you add the [] on BILL to follow the style of the other commands?

File: src/main/java/seedu/guest/model/guest/Bill.java [hunk 2]
```diff
@@ -9,9 +9,12 @@
  */
 public class Bill {
     public static final String MESSAGE_CONSTRAINTS =
-            "Bill should only contain numerical digits and a decimal point, and it should be a non-negative value.\n"
-                    + "If a decimal point is used, there must be 2 digits after it (e.g., 9.99)";
-    public static final String VALIDATION_REGEX = "^\\d+(.\\d{2})?$";
+            "Bills should only contain numerical digits, one decimal point, and one hyphen or plus sign.\n"
+                    + "If a decimal point is used, there must be 2 digits after it (e.g., 9.99).\n"
```
Comment (jehousoh) [inline src/main/java/seedu/guest/model/guest/Bill.java]: Should we allow 0 or 1 digit to be more flexible?

File: src/test/java/seedu/guest/logic/commands/BillCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+        assertCommandFailure(billCommand, model, Messages.MESSAGE_INVALID_GUEST_DISPLAYED_INDEX);
+    }
+
+    /**
+     * Bill from filtered list where index is larger than size of filtered list,
+     * but smaller than size of guest book
+     */
+    @Test
+    public void execute_invalidGuestIndexFilteredList_failure() {
```
Comment (jehousoh) [inline src/test/java/seedu/guest/logic/commands/BillCommandTest.java:169]: Should we add a test for negativeBill as well?

**Review by jehousoh — CHANGES_REQUESTED** [review]
Left some comments do let me know what you think! Good work~

File: docs/UserGuide.md (hunk 1 above)
Comment (JustinPeng13) [reply docs/UserGuide.md:130]: The `[]` are for optional tags, so they wouldn't apply to the bill command as the input value is required. You may refer to the add vs edit command for the difference

File: src/main/java/seedu/guest/model/guest/Bill.java (hunk 2 above)
Comment (JustinPeng13) [reply src/main/java/seedu/guest/model/guest/Bill.java]: Agreed. Also accepted no leading digit before decimal point

File: src/test/java/seedu/guest/logic/commands/BillCommandTest.java (hunk 3 above)
Comment (JustinPeng13) [reply src/test/java/seedu/guest/logic/commands/BillCommandTest.java:169]: Yes, added

**Review by jehousoh — APPROVED** [review]
LGTM!

**Pull request merged**